javascript/ql/src/experimental/Security/CWE-347/JWTMissingSecretOrPublicKeyVerification.ql

```diff
@@ -0,0 +1,19 @@
+/**
+ * @name JWT Missing Secret Or Public Key Verification
+ * @description The software does not verify the JWT token with a cryptographic secret or public key.
+ * @kind problem
+ * @problem.severity warning
+ * @precision high
+ * @id js/jwt-missing-secret-or-public-key-verification
+ * @tags security
+ *       external/cwe/cwe-347
+ */
+
+import javascript
+import DataFlow
+
+from CallNode call
+where
+  call = moduleMember("jsonwebtoken", "verify").getACall() and
+  call.getArgument(1).analyze().getABooleanValue() = false
```

Comment:
```suggestion
  unique(boolean b | b = call.getArgument(1).analyze().getABooleanValue()) = false
```

If our type-inference cannot infer a value for an expression, then it will infer that the expression can have any value.  
I think you can gain some precision by requiring that the only possible boolean value is `false`. 

You can see the difference by comparing [this query run](https://lgtm.com/query/3813952752945038882/) with [this query run](https://lgtm.com/query/1743844830856878629/).

Reply:
Awesome! 👍 
I'm still learning a lot about CodeQL.